improver/spotdata/spot_extraction.py

```diff
@@ -315,9 +320,17 @@ def process(
         spot_values = diagnostic_cube.data[..., y_indices, x_indices]
 
         additional_dims = None
+        additional_dims_aux = []
         if len(spot_values.shape) > 1:
             additional_dims = diagnostic_cube.dim_coords[:-2]
-
+            for dim_coord in additional_dims:
+                dim_coord_dim = diagnostic_cube.coord_dims(dim_coord)
+                aux_coords = [
+                    aux_coord
+                    for aux_coord in diagnostic_cube.aux_coords
+                    if diagnostic_cube.coord_dims(aux_coord) == dim_coord_dim
+                ]
+                additional_dims_aux.append(aux_coords if aux_coords else [])
```

Comment:
This is not unit tested. Are you relying on the acceptance test?

Reply:
Yes, I think I am. Probably not ideal. Would you like me to write a unit test for this specific case?